plugin/index.js

```diff
@@ -263,16 +182,153 @@ class DashboardPlugin {
         },
         complete: this.cleanup
       });
     });
   }
+
+  /**
+   * Infer the root of the project, w/ package.json + node_modules.
+   *
+   * Inspectpack's `version` option needs to know where to start resolving
+   * packages from to translate `~/lodash/index.js` to
+   * `/ACTUAL/PATH/node_modules/index.js`.
+   *
+   * In common practice, this is _usually_ `bundle.context`, but sometimes folks
+   * will set that to a _different_ directory of assets directly copied in or
+   * something.
+   *
+   * To handle varying scenarios, we resolve the project's root as:
+   * 1. Plugin `root` option, if set
+   * 2. `bundle.context`, if `package.json` exists
+   * 3. `process.cwd()`, if `package.json` exists
+   * 4. `null` if nothing else matches
+   *
+   * @param {Object} bundle Bundle
+   * @returns {String|null} Project root path or null
+   */
+  getProjectRoot(bundle) {
+    /*eslint-disable global-require*/
+    // Start with plugin option (and don't check it).
+    // We **will** allow a bad project root to blow up webpack-dashboard.
+    if (this.root) {
+      return this.root;
+    }
+
+    // Try bundle context.
+    try {
+      if (bundle.context && require(path.join(bundle.context, "package.json"))) {
+        return bundle.context;
+      }
+    } catch (err) { /* passthrough */ }
+
+    // Try CWD.
+    try {
+      if (require(path.resolve("package.json"))) {
+        return process.cwd();
+      }
+    } catch (err) { /* passthrough */ }
+
+    // A null will be filtered out, disabling `versions` action.
+    return null;
+  }
+
+  observeBundleMetrics(stats, inspectpack) {
+    const bundlesToObserve = Object.keys(stats.compilation.assets)
+      .filter(
+        bundlePath =>
+          // Don't include hot reload assets, they break everything
+          // and the updates are already included in the new assets
+          bundlePath.indexOf(".hot-update.") === -1 &&
+          // Don't parse sourcemaps!
+          path.extname(bundlePath) === ".js"
+      )
+      .map(bundlePath => ({
+        path: bundlePath,
+        context: stats.compilation.options.context,
+        source: stats.compilation.assets[bundlePath].source()
+      }));
+
+    const getSizes = bundles =>
+      Promise.all(
+        bundles.map(bundle =>
+          inspectpack
+            .sizes({
+              code: bundle.source,
+              format: "object",
+              minified: true,
+              gzip: true
+            })
+            .then(metrics => ({
+              path: bundle.path,
+              metrics
+            }))
+        )
+      )
+        .then(bundle => ({
+          type: "sizes",
+          value: bundle
+        }))
+        .catch(err => ({
+          type: "sizes",
+          error: true,
+          value: serializeError(err)
+        }));
+
+    const getProblems = bundles =>
+      Promise.all(
+        INSPECTPACK_PROBLEM_ACTIONS.map(action =>
+          Promise.all(
+            bundles
```

Comment:
I've actually changed stuff in L288-L313 -- testing project root and removing `versions` if there isn't a package.json